delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/ElasticSearchViews.scala

```diff
@@ -0,0 +1,130 @@
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewState.{Current, Initial}
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewValue._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model._
+import ch.epfl.bluebrain.nexus.delta.sdk.Permissions
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.IOUtils
+import monix.bio.{IO, UIO}
+
+class ElasticSearchViews {}
+
+object ElasticSearchViews {
+
+  /**
+    * The elasticsearch module type.
+    */
+  final val moduleType: String = "elasticsearchview"
+
+  private[elasticsearch] def next(
+      state: ElasticSearchViewState,
+      event: ElasticSearchViewEvent
+  ): ElasticSearchViewState = {
+    // format: off
+    def created(e: ElasticSearchViewCreated): ElasticSearchViewState = state match {
+      case Initial     => Current(e.id, e.project, e.value, e.source, Map.empty, e.rev, deprecated = false,  e.instant, e.subject, e.instant, e.subject)
+      case s: Current  => s
+    }
+
+    def updated(e: ElasticSearchViewUpdated): ElasticSearchViewState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, value = e.value, source = e.source, updatedAt = e.instant, updatedBy = e.subject)
+    }
+
+    def tagAdded(e: ElasticSearchViewTagAdded): ElasticSearchViewState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, tags = s.tags + (e.tag -> e.targetRev), updatedAt = e.instant, updatedBy = e.subject)
+    }
+    // format: on
+
+    def deprecated(e: ElasticSearchViewDeprecated): ElasticSearchViewState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, deprecated = true, updatedAt = e.instant, updatedBy = e.subject)
+    }
+
+    event match {
+      case e: ElasticSearchViewCreated    => created(e)
+      case e: ElasticSearchViewUpdated    => updated(e)
+      case e: ElasticSearchViewTagAdded   => tagAdded(e)
+      case e: ElasticSearchViewDeprecated => deprecated(e)
+    }
+  }
+
+  private[elasticsearch] def evaluate(permissions: UIO[Permissions])(
+      state: ElasticSearchViewState,
+      cmd: ElasticSearchViewCommand
+  )(implicit clock: Clock[UIO] = IO.clock): IO[ElasticSearchViewRejection, ElasticSearchViewEvent] = {
+
+    def validate(value: ElasticSearchViewValue): IO[ElasticSearchViewRejection, Unit] =
+      value match {
+        case _: AggregateElasticSearchViewValue => IO.unit
+        case v: IndexingElasticSearchViewValue  =>
+          for {
+            perms <- permissions
+            set   <- perms.fetchPermissionSet
+            _     <- if (set.contains(v.permission)) IO.unit else IO.raiseError(PermissionIsNotDefined(v.permission))
+          } yield ()
+      }
+
+    def create(c: CreateElasticSearchView) = state match {
+      case Initial =>
+        for {
+          _ <- validate(c.value)
```

Comment:
On creation we might want to validate if the ES index already exists. But since the indices are UUID values, maybe that's not necessary

Reply:
I think the management of indices is part of the indexing logic; I would not put it here.